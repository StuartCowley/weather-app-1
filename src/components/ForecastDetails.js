import React from "react";
import PropTypes from "prop-types";
import WeatherIcon from "react-icons-weather";
import moment from "moment";

const ForecastDetails = ({ forecast }) => {
  const { date, description, icon, temperature, humidity, wind } = forecast;
  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__date">
        {moment(date).format("ddd Do MMM")}
      </div>
      <div className="forecast-details__icon" data-testid="forecast-icon">
        <WeatherIcon name="owm" iconId={icon.toString()} />
      </div>
      <div
        className="forecast-details__temperature"
        data-testid="forecast-temperature"
      >
        Max:{temperature.max}&#176;C
        <br />
        Min:{temperature.min}&#176;C
      </div>
      <div className="forecast-details__description">{description}</div>
      <div className="forecast-details__humidity">{humidity}</div>
      <div className="forecast-details__wind">
        {wind.speed}
        {wind.direction}
      </div>
    </div>
  );
};

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number,
    temperature: PropTypes.shape({
      min: PropTypes.number,
      max: PropTypes.number,
    }).isRequired,
    description: PropTypes.string.isRequired,
    icon: PropTypes.number.isRequired,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }).isRequired,
    humidity: PropTypes.number.isRequired,
  }),
};

ForecastDetails.defaultProps = {
  forecast: PropTypes.shape({
    date: 0,
    temperature: PropTypes.shape({
      min: -60,
      max: 60,
    }),
    description: "",
    icon: 800,
    wind: PropTypes.shape({
      speed: 0,
      direction: "",
    }),
    humidity: 0,
  }),
};

export default ForecastDetails;
