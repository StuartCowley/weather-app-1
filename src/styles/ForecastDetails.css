.forecast-details {
  border: solid;
  border-color: #66748a;
  background-color: #aaafb9;
  border-radius: 25px;
  margin: 0 10% 0 10%;
  padding: 5%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}
