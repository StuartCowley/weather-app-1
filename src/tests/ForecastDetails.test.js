import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProp = {
    date: 1525132800000,
    icon: 800,
    temperature: {
      min: 12,
      max: 22,
    },
    humidity: 40,
    wind: {
      speed: 5,
      direction: "ne",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProp} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly", () => {
    const { getByText, getByTestId } = render(
      <ForecastDetails forecast={validProp} />
    );

    expect(getByTestId("forecast-icon")).toHaveClass("forecast-details__icon");
    expect(getByText("Tue 1st May")).toHaveClass("forecast-details__date");

    expect(getByTestId("forecast-temperature")).toHaveClass(
      "forecast-details__temperature"
    );
    expect(getByText("40%")).toHaveClass("forecast-details__humidity");
    expect(getByTestId("forecast-wind-details")).toHaveClass(
      "forecast-details__wind"
    );
  });
});
