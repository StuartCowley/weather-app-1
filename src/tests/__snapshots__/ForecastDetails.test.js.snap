// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Tue 1st May
    </div>
    <div
      class="forecast-details-bar"
    >
      <div
        class="forecast-details__description"
      >
        stubDescription
      </div>
      <div
        class="forecast-details__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-sunny"
          name="owm"
        />
      </div>
      <div
        class="forecast-details__temperature"
        data-testid="forecast-temperature"
      >
        Max:22°C
        <br />
        Min:12°C
      </div>
      <div
        class="forecast-details__wind"
      >
        <div
          class="forecast-details__humidity"
        >
          40%
        </div>
        <div
          class="wind-details"
        >
          5ms
          <img
            alt="img"
            src="arrow.svg"
            style="transform: rotate(45deg);"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
